File: sdkwork-ai-sdk-java/src/main/java/com/sdkwork/ai/api/Audio2Api.java
package com.sdkwork.ai.api;

import com.sdkwork.ai.http.HttpClient;
import com.sdkwork.ai.model.*;
import java.util.List;
import java.util.Map;

public class Audio2Api {
    private final HttpClient client;
    
    public Audio2Api(HttpClient client) {
        this.client = client;
    }

    /** Create translation */
    public TranscriptionResponse createTranslation(CreateTranslationPostRequest body, Map<String, Object> params) throws Exception {
        return (TranscriptionResponse) client.post(ApiPaths.aiPath("/v1/audio/translations"), body, params, null, "multipart/form-data");
    }

    /** Create transcription */
    public TranscriptionResponse createTranscription(CreateTranscriptionRequest body, Map<String, Object> params) throws Exception {
        return (TranscriptionResponse) client.post(ApiPaths.aiPath("/v1/audio/transcriptions"), body, params, null, "multipart/form-data");
    }

    /** Create speech */
    public String createSpeech(SpeechRequest body) throws Exception {
        return (String) client.post(ApiPaths.aiPath("/v1/audio/speech"), body);
    }
}

File: sdkwork-ai-sdk-kotlin/src/main/kotlin/com/sdkwork/ai/api/Audio2Api.kt
package com.sdkwork.ai.api

import com.sdkwork.ai.*
import com.sdkwork.ai.http.HttpClient

class Audio2Api(private val client: HttpClient) {

    /** Create translation */
    suspend fun createTranslation(body: CreateTranslationPostRequest? = null, params: Map<String, Any>? = null): TranscriptionResponse? {
        return client.post(ApiPaths.aiPath("/v1/audio/translations"), body, params, null, "multipart/form-data") as? TranscriptionResponse
    }

    /** Create transcription */
    suspend fun createTranscription(body: CreateTranscriptionRequest? = null, params: Map<String, Any>? = null): TranscriptionResponse? {
        return client.post(ApiPaths.aiPath("/v1/audio/transcriptions"), body, params, null, "multipart/form-data") as? TranscriptionResponse
    }

    /** Create speech */
    suspend fun createSpeech(body: SpeechRequest): String? {
        return client.post(ApiPaths.aiPath("/v1/audio/speech"), body) as? String
    }
}
